### PR TITLE
editEntities - select all in/touching box

### PR DESCRIPTION
**Pull request by Atlante45**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 55a88e60764cca1066d05e4386163beec0cfd423 available at: https://gist.github.com/e9882e3e8ceae0b03aac<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4442.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4442.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 36814e837b050b20de9e4d30bdecf17f7d2671fd available at: https://gist.github.com/e9882e3e8ceae0b03aac<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4442.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4442.dmg

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 1]
```diff
@@ -208,7 +208,24 @@ QVector<EntityItemID> EntityScriptingInterface::findEntities(const glm::vec3& ce
         QVector<const EntityItem*> entities;
         _entityTree->findEntities(center, radius, entities);
         _entityTree->unlock();
+        
+        foreach (const EntityItem* entity, entities) {
+            EntityItemID thisEntityItemID(entity->getID(), UNKNOWN_ENTITY_TOKEN, true);
```
Comment (ZappoMan) [inline libraries/entities/src/EntityScriptingInterface.cpp]: why not just use:
```
result << entity->getEntityItemID();
```

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 2]
```diff
@@ -208,7 +208,24 @@ QVector<EntityItemID> EntityScriptingInterface::findEntities(const glm::vec3& ce
...
+    if (_entityTree) {
+        _entityTree->lockForRead();
+        AABox box(corner, dimensions);
+        QVector<EntityItem*> entities;
+        _entityTree->findEntities(box, entities);
+        _entityTree->unlock();
+        
         foreach (const EntityItem* entity, entities) {
             EntityItemID thisEntityItemID(entity->getID(), UNKNOWN_ENTITY_TOKEN, true);
             result << thisEntityItemID;
```
Comment (ZappoMan) [inline libraries/entities/src/EntityScriptingInterface.cpp]: same here... should be... 

```
result << entity->getEntityItemID();
```

**Comment by hifi-gustavo** [comment]
Build console output for commit: c6167f689348e2812da4e3388699cb4db043ff25 available at: https://gist.github.com/e9882e3e8ceae0b03aac<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4442.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4442.dmg

File: examples/edit.js [hunk 3]
```diff
@@ -888,8 +933,15 @@ function handeMenuEvent(menuItem) {
...
         entityListTool.toggleVisible();
+<<<<<<< HEAD
+    } else if (menuItem == "Select All Entities In Box") {
+        selectAllEtitiesInCurrentSelectionBox(false);
+    } else if (menuItem == "Select All Entities Touching Box") {
+        selectAllEtitiesInCurrentSelectionBox(true);
+=======
     } else if (menuItem == MENU_SHOW_LIGHTS_IN_EDIT_MODE) {
         lightOverlayManager.setVisible(isActive && Menu.isOptionChecked(MENU_SHOW_LIGHTS_IN_EDIT_MODE));
+>>>>>>> dfd39ca1e14bb2198b12374b6bc863b3c27f673c
```
Comment (AndrewMeadows) [inline examples/edit.js]: There is an unresolved merge conflict here.  I wonder why the automatic build system gives this PR a green checkmark.  Maybe github is confused?  I'll try to clone and build manually.

Comment (Atlante45) [inline examples/edit.js]: Oh ok, that's why xcode didn't complain.
Thanks, fixed.

**Comment by hifi-gustavo** [comment]
Build console output for commit: c421adaf5882eb8f4833457369c7e6f8b72542f0 available at: https://gist.github.com/e9882e3e8ceae0b03aac<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4442.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4442.dmg

**Pull request merged**